NEXT-32619 - Write document for order approval (#1240)
shopware/docs: products/extensions/b2b-components/order-approval/01-entities-and-workflow.md

### Pending Order

The pending order entity represents an order that has been placed by an employee that requires approval. It contains the order's data, the employee that placed the order and the approval rule that matched the order.

## Workflow

The following diagram shows the workflow of the order approval component:

```mermaid
flowchart TD
    A(Employee places an order) -->B{Approval rule applies}
    B-->|No| C[Event: Order Placed]
    B-->|Yes| D[Event: Order needs approval]
    D-->E{Order approved?}
    E-->|No| F[Event: Order declined]
    E-->|Yes| G[Event: Order Approved & Event: Order placed]
```

## Who can request approval?

> - Employees which have the role that is assigned as the "Effective role" of the approval rule that matched the order.

Reviewer: ```suggestion
* Employees holding the role designated as the "Effective role" in the approval rule corresponding to the order are authorized to request approval.
```

## Who can view pending orders?

- Employees with the "Can view all pending orders" permission can view all pending orders.

> - Employees which requested approval for the order can view his pending orders.

Reviewer: ```suggestion
- Employees who requested approval for the order can view their pending orders.
```

- Business Partners can view all pending orders of their employees.

> ## Who can approve/decline pending orders?

Reviewer: ```suggestion
## Who can approve or decline pending orders?
```

> - Employees with the "Can approve/decline all pending orders" permission can approve/decline all pending orders.
> - Employees with the "Can approve/decline pending orders" permission can approve/decline pending orders that assigned to him.
> - Business Partners can approve/decline all pending orders of their employees.

Reviewer: ```suggestion
- Employees with the "Can approve/decline all pending orders" permission can approve or decline all pending orders.
- Employees with the "Can approve/decline pending orders" permission can approve or decline pending orders that assigned to them.
- Business Partners can approve or decline all pending orders of their employees.
```
